app/scripts/modules/core/pipeline/config/stages/destroyAsg/aws/awsDestroyAsgStage.js

```diff
@@ -38,18 +39,37 @@ module.exports = angular.module('spinnaker.core.pipeline.stage.aws.destroyAsgSta
       regionsLoaded: false
     };
 
+    let clusterFilter = (cluster) => {
+      let acctFilter = $scope.stage.credentials ? cluster.account === $scope.stage.credentials : true;
+      let regionFilter = $scope.stage.regions && $scope.stage.regions.length
+        ? _.some( cluster.serverGroups, (sg) => _.some($scope.stage.regions, (region) => region === sg.region))
+        : true;
+
+      return acctFilter && regionFilter;
+    };
```

Comment:
Looks like this is repeated in eight different places. Is there a way to consolidate it into a service that each cloud provider can implement? Otherwise, GCE, CloudFoundry, Azure, and Titan stages will all do the same thing and we'll end up with this block (or a very similar block) repeated a bunch of times.